pkg/k8s/watcher_linux.go

```diff
@@ -92,3 +103,18 @@ func Start(ctx context.Context, k *watchers.K8sWatcher) {
 	<-syncdCache
 	logger.Info("Kubernetes watcher synced")
 }
+
+// retinaK8sErrorHandler is a custom error handler for the watcher
+// that logs the error and tags the error to easily identify
+func k8sWatcherErrorHandler(e error) {
+	errStr := e.Error()
+	switch {
+	case strings.Contains(errStr, "Failed to watch *v1.Node"):
+	case strings.Contains(errStr, "Failed to watch *v2.CiliumEndpoint"):
+	case strings.Contains(errStr, "Failed to watch *v1.Service"):
+	case strings.Contains(errStr, "Failed to watch *v2.CiliumNode"):
+		logger.WithField("underlyingError", errStr).Error("Error watching k8s resource")
```

Comment:
I dont think we should combine per resource errors together, I would highly suggest to keep them insividual because sometimes if there is an issue with one of the component, we should know which one has caused it.

Reply:
I can add a firld to tell the resource type.